src/Utils.ts

```diff
@@ -29,7 +29,8 @@ export function splitLines(str: string): string[] {
  * @return {RegExpMatchArray} The resulting match, if any.
  */
 export function matchStart(str: string, regexp: RegExp): RegExpMatchArray {
-  let match: any = str.match(regexp);
+  const startRegexp = new RegExp(`^${regexp.source}`, regexp.flags);
+  let match: any = str.match(startRegexp);
```

Comment:
Looks safe to me. Mainly curiosity: why does reconstructing the regexp improve performance here?

Reply:
The added leading carrot ^ prevents the regexp from matching in the middle of a string. 

Note that this doesn't fix all potential issues with our regexes.  See [this example](https://snyk.io/vuln/SNYK-JS-MARKDOWNIT-459438) for some other fun examples of inefficient regexes (we should upgrade that too)

FWIW, I believe it's safe to have two carrots in a row and it's generally not possible to robustly detect if there is already a leading carrot (or the equivalent, since parenthesis could be involved).